Makefile

```diff
@@ -17,7 +17,11 @@
 ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 
 # Base path used to install.
-DESTDIR ?= /usr/local
```

Comment:
This change is likely to break downstream projects.
Can we keep `DESTDIR` for compatibility?

Reply:
Which ones ? Do you know any specific ones, so we could talk to them ?

I've did some web research and the only downstream I've found that explicitly builds containerd and sets this variable
is the meanwhile dead docker-ce repo. And this happened in their own packaging rule files, which - from a distro packager perspective - are completely broken and drive the whole process in the exactly wrong direction:

wrong:
* upstream source has some make rule that directly produces .deb / .rpm files, buypassing the distro infrastructure.
correct: 
* upstream just provides build scripts that follow the common standards, allows build time customization (pathes, toolchain, ...) and just drive build process until the install stage ('make install' into a temporary root), but NOT attempting to do anything distro specific
* distro adds their build stuff (some, eg. gentoo, have that even completely separate from the source tree), finally driving the upstream's build process from *within* *their* build infrastructure (eg. buildd, pbuilder, dck-buildpackage, ...), which is also the one that creates installable packages from the temporary root image.
* anything that's distro/site specific is injected by the distro build infrastructure - upstream's build system should never even try to guess anything here.

Since docker-ce seems officially dead, we IMHO can ignore it completely. So, the only remaining candidates I personally know of are the Distros - exactly those who need to carry their own patches, because the *current* way *does* break them. (I happen to be one of these guys).

One of the most time-consuming tasks for us distro maintainers / packagers is upstream's fixing broken build systems.
And quite frankly, we're pretty sick of that. And that's one reason why distros sometimes lack so far behind upstream releases. (and for many professional applications, using distro packages is pretty much mandatory).

The way this variable is used here, indeed *is* broken, as since about 30+ years, that variable is meant to specify a 
temporary target root directory (that only applies when actually copying the files into a target image), that has nothing to do with $PREFIX.

Note that $DESTDIR's content may never appear in the actually installed / packaged file. It if does, that package is just broken.

I can *imagine* that Docker Inc. still uses the broken current approach internally - I've got no idea, since it's kept secret.
If that's the case, then I'd highly recommend fixing their build infrastructure by doing it by the distro's official processes.
The way it's done in the now dead docker-ce is just error prone, expensive and conceptionally broken.

If anybody helps help w/ proper distro packaging, feel free to contact me directly.

--mtx

Reply:
> Which ones ? Do you know any specific ones, so we could talk to them ?

Lots of users:
- https://github.com/termux/termux-root-packages/blob/24d4fa6855169dd76ae5af4e3359d4838fcdc7d1/packages/containerd/build.sh#L32
- https://github.com/vmware/photon/blob/2cf76b75a32d4ff0c578f5f67a4ef4a2b9c1f3d6/SPECS/containerd/containerd.spec#L66
- https://github.com/fusion809/void-packages-bleed/blob/18b951b3354f572d24eb656304d51027f21a6752/srcpkgs/containerd/template#L26
https://github.com/yichu77/packages/blob/48ac138689cf6fccd3d02758d0d07c18cf55adcb/utils/containerd/Makefile#L54

Also, probably there would be a bunch of private scripts that use `DESTDIR`.

> Since docker-ce seems officially dead, we IMHO can ignore it completely.

I don't understand what you are talking about.

Reply:
> > Which ones ? Do you know any specific ones, so we could talk to them ?
> 
> Lots of users:
> 
> * https://github.com/termux/termux-root-packages/blob/24d4fa6855169dd76ae5af4e3359d4838fcdc7d1/packages/containerd/build.sh#L32
> * https://github.com/vmware/photon/blob/2cf76b75a32d4ff0c578f5f67a4ef4a2b9c1f3d6/SPECS/containerd/containerd.spec#L66
> * https://github.com/fusion809/void-packages-bleed/blob/18b951b3354f572d24eb656304d51027f21a6752/srcpkgs/containerd/template#L26
>   https://github.com/yichu77/packages/blob/48ac138689cf6fccd3d02758d0d07c18cf55adcb/utils/containerd/Makefile#L54
> 

Okay, these are distros, who use specific releases - and they had to do extra magic to cope with that very exotic
use of $DESTDIR (note that my patch just aligns it to what is common practise since 30 years).

If they, some day, move up to some newer major release, intensive testing is needed anyways, and can easily remove the non-standard logic. I'll contact them for preparation ...

> Also, probably there would be a bunch of private scripts that use `DESTDIR`.

I guess it would be the same for them. And I really don't believe that trying to cope with somebody's internal hacks,
which we won't even ever see, can be any basis for decent engineering.

> > Since docker-ce seems officially dead, we IMHO can ignore it completely.
> 
> I don't understand what you are talking about.

See: https://github.com/docker/docker-ce

"This repository is now deprecated and will be archived"

That sounds like docker-ce is discontinued, and us package maintainers are supposed to directly put together the
docker stack from the individual upstreams. (exactly what I'm doing right now).

Is there a new location for the docker-ce git repo ?

--mtx

Reply:
The docker-ce repo was deprecated, but docker-ce product is alive.
Built from https://github.com/moby/moby https://github.com/docker/cli https://github.com/docker/docker-packaging https://github.com/docker/containerd-packaging

Reply:
![bart-distro](https://user-images.githubusercontent.com/46083/117354062-35ccf100-aeb1-11eb-8ca5-ad75a94139a6.png)

Reply:
Where is `DESTDIR` being set now?